articles/azure-functions/functions-reference-powershell.md

```diff
@@ -0,0 +1,424 @@
+
+ms.assetid: 45dedd78-3ff9-411f-bb4b-16d29a11384c
+ms.service: azure-functions
+ms.devlang: powershell
+ms.topic: reference
+ms.date: 11/5/2018
+ms.author: tyleonha
+
+---
+# Azure Functions PowerShell developer guide
+
+This guide contains information about the intricacies of writing Azure Functions with PowerShell.
+
+A PowerShell function is represented as a PowerShell script that executes when triggered ([triggers are configured in function.json](functions-triggers-bindings.md)). The powershell script takes in parameters that match the names of all the input bindings. In addition to those inputs, a parameter is available to you called `TriggerMetadata` that contains additional information on the trigger that started the function.
+
+This article assumes that you have already read the [Azure Functions developer reference](functions-reference.md). You should also complete the Functions quickstart to create your first function, using [Visual Studio Code](functions-create-first-function-vs-code.md) or [via the Azure Functions Core Tools](functions-create-first-azure-function.md).
+
+## Folder structure
+
+The required folder structure for a JavaScript project looks like the following. This default can be changed. For more information, see the [scriptFile](#using-scriptfile) section below.
+
+```
+PSFunctionApp
+ | - MyFirstFunction
+ | | - run.ps1
+ | | - function.json
+ | - MySecondFunction
+ | | - run.ps1
+ | | - function.json
+ | - Modules
+ | | - myFirstHelperModule.psm1
+ | | - mySecondHelperModule.psm1
+ | - local.settings.json
+ | - host.json
+ | - extensions.csproj
+ | - bin
+```
+
+At the root of the project, there's a shared [host.json](functions-host-json.md) file that can be used to configure the function app. Each function has a folder with its own code file (.ps1) and binding configuration file (function.json). The name of `function.json`'s parent directory is always the name of your function.
+
+The binding extensions required in [version 2.x](functions-versions.md) of the Functions runtime are defined in the `extensions.csproj` file, with the actual library files in the `bin` folder. When developing locally, you must [register binding extensions](functions-triggers-bindings.md#local-development-azure-functions-core-tools). When developing functions in the Azure portal, this registration is done for you.
+
+## Defining a PowerShell script to be a function
+
+By default, the Functions runtime looks for your function in `run.ps1`, where `run.ps1` shares the same parent directory as its corresponding `function.json`.
+
+Your script is passed a number of arguments on execution. The powershell script takes in parameters that match the names of all the input bindings. In addition to those inputs, a parameter is available to you called `TriggerMetadata` that contains additional information on the trigger that started the function. You can define this at the top of your script by adding a `param` block:
```

Comment:
Seems like there should be a section or another doc that explains the members of `$TriggerMetadata`

Reply:
added a section on TriggerMetadata